src/backend/BSFileDB.pm

```diff
@@ -59,7 +65,7 @@ sub encode_line {
   ($al) = splice(@lay, -2, 2) if $lay->[-1] eq '[]';
   push @line, defined($r->{$_})  ? $r->{$_} : '' for @lay;
   push @line, @{$r->{$al} || []} if $al;
-  s/([\000-\037%|=\177-\237])/sprintf("%%%02X", ord($1))/ge for @line; 
+  @line = map {encode_field($_)} @line;
```

Comment:
Please don't call encode_field here. Perl is a bit slow in regard to function calls, unlike languages like C. Just leave the code duplicated.

Reply:
okay dropped the change